src/classes/SObjectRepository.cls

```diff
@@ -2,40 +2,98 @@
 * This file is part of the Nebula Framework project, released under the MIT License.             *
 * See LICENSE file or go to https://github.com/jongpie/NebulaFramework for full license details. *
 *************************************************************************************************/
-public abstract class SObjectRepository extends NebulaCore implements ISObjectRepository {
+public virtual class SObjectRepository extends NebulaCore implements ISObjectRepository, ICrud {
 
+    private final Schema.SObjectType sObjectType;
+    private final Schema.FieldSet fieldSet;
+    private final List<Schema.SObjectField> sobjectFieldList;
+    private final Schema.SObjectField idField;
     protected IQueryBuilder queryBuilder;
 
-    protected SObjectRepository(Schema.SObjectType sobjectType, Schema.FieldSet fieldSet) {
+    public SObjectRepository(Schema.SObjectType sobjectType, Schema.FieldSet fieldSet) {
         this(sobjectType, fieldSet, null);
     }
 
-    protected SObjectRepository(Schema.SObjectType sobjectType, List<Schema.SObjectField> sobjectFieldList) {
+    public SObjectRepository(Schema.SObjectType sobjectType, List<Schema.SObjectField> sobjectFieldList) {
         this(sobjectType, null, sobjectFieldList);
     }
 
     private SObjectRepository(Schema.SObjectType sobjectType, Schema.FieldSet fieldSet, List<Schema.SObjectField> sobjectFieldList) {
         this.currentModule = NebulaCore.Module.REPOSITORY;
 
-        this.queryBuilder = new QueryBuilder(sobjectType, fieldSet, sobjectFieldList);
```

Comment:
removed in favor of line 30, below - fixed a nasty bug where the query builder was saving state if multiple calls to the repository was made in the same transaction